### DISCUSSION TITLE
How do I use TCP in bevy?

### DISCUSSION THREAD

**Discussion by thebluetropics** [bevyengine/bevy#11665] [Q&A]
I am new to Bevy and Rust. I have made several working web apps (using `actix-web`) and a dedicated game server (using non-blocking sychronous TCP from the standard library)

However, at this point, I have no clue how to make a simple TCP connection (client-server) in a bevy program. The only thing in my head is polling, on each `Update`.

If possible, I need an example, I haven't found it anywhere.

**Comment by BD103** [comment]
Hi! I recommend checking out the [Bevy Assets category for networking](https://bevyengine.org/assets/#networking), you may find a library that fits your needs instead of having to make your own. (Make sure to find a library that supports the latest release of Bevy, 0.12.1. There are a few outdated ones!)

I hope that helps :)

**Comment by rlidwka** [comment]
Polling on each update is fine, I've done that before.

Having TCP connection in a separate AsyncComputeTaskPool and communicating through channels is another idea.

**Reply by thebluetropics** [reply]
Hi @rlidwka, thank you for sharing the idea.

I haven't understand how things are work each other, I only have a bit experience in ECS.

Where should I store the `TcpStream` or `TcpListener`? Should I store it as a resource or entity with custom component?

**Reply by rlidwka** [reply]
I've stored it in a resource, but then moved to entity once I've realized that I would have multiple tcp connections.
